packages/hooks/src/useEventListener.ts

```diff
@@ -17,17 +17,21 @@ interface UseEventOptions {
  * @param optional.dependencies Array to be passed to useEffect hook, such that the hook will only run if the array's values have changed.
  * @param optional.element Value to be passed as target of event handler, will default to document.
  */
-export default function useEventListener(
-  type: keyof GlobalEventHandlersEventMap,
-  eventCallback: (e) => void,
+export default function useEventListener<
+  Type extends keyof GlobalEventHandlersEventMap
+>(
+  type: Type,
+  eventCallback: (e: GlobalEventHandlersEventMap[Type]) => void,
```

Comment:
If we type this as an event listener, rather than typing the event parameter, can we omit typing it's usage on line 54?

Reply:
We could, but then we would lose the information that, say, a `click` handler is going to get a `MouseEvent`, with `clientX` and `clientY` parameters. It's weird, because `addEventListener` is supposed to do this same mapping, but it seems like it's not following along quite right, something to do with `element` being `Document | Element` rather than just one or the other I think.